include/extractor/restriction_parser.hpp

```diff
@@ -43,15 +45,16 @@ class RestrictionParser
   public:
     RestrictionParser(bool use_turn_restrictions,
                       bool parse_conditionals,
-                      std::vector<std::string> &restrictions);
+                      const std::vector<std::string> &restrictions);
     std::vector<InputTurnRestriction> TryParse(const osmium::Relation &relation) const;
 
   private:
     bool ShouldIgnoreRestriction(const std::string &except_tag_string) const;
 
     bool use_turn_restrictions;
     bool parse_conditionals;
-    std::vector<std::string> restrictions;
+    std::set<std::string> restrictions;
+    osmium::tags::KeyFilter filter;
```

Comment:
The Osmium docs say this is [deprecated](https://docs.osmcode.org/libosmium/latest/namespaceosmium_1_1tags.html#a75df2aa39d225085eeffd438b701bfa1), and should be replaced with `osmium::TagsFilter`.